### PR TITLE
Feature/Redux Reorganizing

### PR DESCRIPTION
**Pull request by MrJacobSullivan**
## Description

This PR breaks up the actions we currently have into individual files and attempts to establish a pattern we can use as our redux store grows. The structure of the initial state object passed into the reducer is also modified to try to maintain a similar structure. I also modified the `HomeContainer` component to account for these structural changes in the state. I also added the two `useEffect` hooks to the `HomeContainer` component that we need in order to establish our initial state after authentication. Eventually, these side effects will go in the authenticated route component that is *WIP*.

[Loom Video](https://www.loom.com/share/4d0351ecf11548fc8072956fb9999d2c)
Trello [#202](https://trello.com/c/t9qdvrKh)
This PR is continuing work from PR [#95 ](https://github.com/BloomTech-Labs/underdog-devs-fe-a/pull/95)

**New State Structure**
```js
const initialState = {
  auth: {
    // Okta will save a JWT in local storage
    // use the Okta library to decipher that JWT and get the user_id
    isAuthenticated: false,
    profile_id: null,
  },
  userProfile: {
    // when user hits dashboard, make API call to [GET] /profiles/:id
    // first_name: 'Test',
    // last_name: 'Test',
    // role_id: 3,
    // ...etc.
  },
  lifecycle: {
    // set to true when API call starts, set to false when API call concludes
    isFetching: false,
  },
  errors: [
    /* TODO: implement a way to track errors */
  ],
};
```

## Contributors
@MrJacobSullivan 
@Jaredhall18 
@ryansnorek 
@ST1414

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have removed unnecessary comments/console logs from my code
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings
- [x] No duplicate code left within changed files
- [x] Size of pull request kept to a minimum
- [x] Pull request description clearly describes changes made & motivations for said changes

### CODE REVIEW

File: src/state/actions/auth/index.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export * from './setProfileId';
```
Comment (ryansnorek) [inline src/state/actions/auth/index.js:1]: Do we need to export *  if the functions and variables are exported inline?

**Review by ryansnorek — APPROVED** [review]
Love the reducer object structure. Simple and clean

File: src/state/actions/auth/index.js (hunk 1 above)
Comment (MrJacobSullivan) [reply src/state/actions/auth/index.js:1]: My intention was to make the action constants, i.e. `SET_USER_PROFILE` easily accessible. For instance `import { SET_USER_PROFILE } from './src/state/actions';` as opposed to `import { SET_USER_PROFILE } from './src/state/actions/userProfile/setUserProfile'`. However, this does add like 6 files to the codebase and might cause some confusion if anyone is trying to find where an action constant is coming from. I'll go ahead and simplify this approach.

**Review by Jaredhall18 — APPROVED** [review]
I agree with your decision to make constants and exporting out of the index file, just keeps imports in other files looking a lot cleaner.  The updated structure for the initial state looks better, thanks for changing that as well.  The loom video was very helpful, thanks for taking the time to walk us through these changes, incredible work.

**Pull request merged**